ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,94 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
             sub_faces.extend(face.sub_faces_by_ratio(ratio))
         return sub_faces
 
+    def _sub_faces_by_dimensions(self, sub_face_height, sub_face_width, wall_length,
+                                dist_breakup, sill_height, rect_height, changed_var):
+        """Get a list of faces with a combined area equal to the glazing ratio
+        times this face area. The glazing ratio is derived internally using the
+        specified sub face height and width along with other parameters.
+
+        All sub faces will lie inside the boundaries of this face.
+
+        Args:
+            height: sub face height
+            width: sub face width
+            wall_length: total length (width) of sub face wall
+            dist_breakup: distance by which main rectangle is meant to be 
+                divided up into points on the bottom
+            sill_height: A number for the target height above the bottom edge of
+                the rectangle to start the sub-rectangles. Note that, if the
+                ratio is too large for the height, the ratio will take precedence
+                and the sub-rectangle height will be smaller than this value.
+            rect_height: A number for the target height of the output rectangles. Note that, if the ratio is too large for the height,
+                the ratio will take precedence and the sub-rectangle height will
+                be larger than this value.
+            changed_var: string indicating the variable being changed 
```

Comment:
I am at a loss to understand what this variable is or why it is important.  I feel like it might be a relic from the Javascript that doesn't really have a purpose here. Could we just remove it from the inputs and always assume the same value for this?

Reply:
In that case, what should I put as the body of the first conditional (Line 1183)? Should it be the line that updates `sub_face_height_final` (Line 1185), or the one that updates `sill_height_final` (Line 1187)?

Reply:
I can see that the code was a relic from the old javascript and it was trying to be sensitive to which input the user had just changed on the UI (so it is not relevant here). Let's give priority to the window height over the sill height so the sill height is what should be adjusted and line 1187 is the one we should use.